packages/compass-generative-ai/src/components/ai-optin-modal.tsx

```diff
@@ -0,0 +1,156 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import {
+  Banner,
+  Body,
+  Link,
+  ConfirmationModal,
+  SpinLoader,
+  css,
+  spacing,
+  H3,
+  palette,
+} from '@mongodb-js/compass-components';
+import { AISignInImageBanner } from './ai-signin-banner-image';
+import { closeOptInModal, optIn } from '../store/atlas-optin-reducer';
+import type { RootState } from '../store/atlas-ai-store';
+import { usePreference } from 'compass-preferences-model/provider';
+
+const GEN_AI_FAQ_LINK = 'https://www.mongodb.com/docs/generative-ai-faq/';
+
+type OptInModalProps = {
+  isOptInModalVisible: boolean;
+  isOptInInProgress: boolean;
+  onOptInModalClose: () => void;
+  onOptInClick: () => void;
+  projectId: string;
+};
+
+const titleStyles = css({
+  marginBottom: spacing[400],
+  marginTop: spacing[400],
+  marginLeft: spacing[500],
+  marginRight: spacing[500],
+  textAlign: 'center',
+});
+
+const bodyStyles = css({
+  marginBottom: spacing[400],
+  marginTop: spacing[400],
+  marginLeft: spacing[300],
+  marginRight: spacing[300],
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'center',
+  textAlign: 'center',
+});
+
+const disclaimer = css({
+  color: palette.gray.dark1,
+  marginTop: spacing[400],
+  marginLeft: spacing[800],
+  marginRight: spacing[800],
+});
+
+const banner = css({
+  padding: spacing[400],
+  marginTop: spacing[400],
+  textAlign: 'left',
+});
+
+const featureStatusBannerVariant = (pIsOrgAiEnabled: boolean) => {
+  switch (true) {
+    case pIsOrgAiEnabled:
+      return 'info';
+    default:
+      return 'warning';
+  }
+};
+
+function getFeatureStatusBannerText(pIsOrgAiEnabled: boolean) {
+  switch (true) {
+    case pIsOrgAiEnabled:
+      return 'AI features are enabled for project users with data access.';
+
+    default:
+      return 'AI features are disabled for project users.';
+  }
```

Comment:
What is this switch for? Here and above. Could we just write this as 
`pIsOrgAiEnabled ? 'AI features are enabled for project users with data access.' : 'AI features are disabled for project users.'`
Looks like this is only used in one place in the component also, I'd suggest not having it as a function and instead having it inline.

Reply:
noted will fix this! was just modeling it off the charts code which did it this way.